lib/jobs/ucs-job.js

```diff
@@ -0,0 +1,195 @@
+di.annotate(ucsJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Util',
+    'Promise',
+    'Services.Waterline',
+    'JobUtils.UcsTool',
+    'Services.Configuration',
+    'Services.Encryption',
+    '_',
+    'Assert'
+));
+
+function ucsJobFactory(
+    BaseJob,
+    Logger,
+    util,
+    Promise,
+    waterline,
+    UcsTool,
+    configuration,
+    encryption,
+    _,
+    assert
+) {
+    var logger = Logger.initialize(ucsJobFactory);
+
+    /**
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function UcsJob(options, context, taskId) {
+        UcsJob.super_.call(this, logger, options, context, taskId);
+        this.routingKey = context.graphId;
+        assert.uuid(this.routingKey);
+
+        this.concurrent = {};
+        this.maxConcurrent = 1;
+        this.ucsComandClassIds = {
+            "ucs.powerthermal": "memoryUnitEnvStats,processorEnvStats," +
+                "computeMbPowerStats,computeMbTempStats,equipmentChassisStats",
+            "ucs.fan": "equipmentFanStats",
+            "ucs.psu": "equipmentPsuStats",
+            "ucs.disk": "storageLocalDiskSlotEp",
+            "ucs.led": "equipmentLed",
+            "ucs.sel": "sysdebugMEpLog"
+        };
+    }
+
+    util.inherits(UcsJob, BaseJob);
+
+    /**
+     * @function _initUcsTool
+     * @description generate a new UcsTool object.
+     **/
+    UcsJob.prototype._initUcsTool = function() {
+        return new UcsTool();
+    };
+
+    /**
+     * @function _run
+     * @description the jobs internal run method
+     **/
+    UcsJob.prototype._run = function() {
+        var self = this;
+
+        return waterline.workitems.update(
+            {name: "Pollers.UCS"}, 
+            {failureCount: 0}
+        )
+        .then(function() {
+            return self._subscribeUcsCommand(
+                self.routingKey,
+                self._subscribeUcsCallback.bind(self)
+            );
+        });
+    };
+
+    UcsJob.prototype._subscribeUcsCallback = function(data) {
+        var self = this;
+        return new Promise(function(reslove, reject) {
+            if (self.concurrentRequests(data.node, data.workItemId)) {
+                return reject(new Error("max number of request has been reached."));
+            } else {
+                self.addConcurrentRequest(data.node, data.workItemId);
+                return reslove();
+            }
+        })
+        .then(function() {
+            return waterline.obms.findByNode(data.node, 'ucs-obm-service', true)
+            .then(function(obmSetting) {
+                if (obmSetting) {
+                    return {
+                        obmSetting: obmSetting,
+                        command: data.config.command,
+                        data: data
+                    };
+                } else {
+                    throw new Error("No ucs-obm-service found for id: %s".format(data.node));
```

Comment:
I would prefer to use promise.reject to make it consist in a promise chain.

Reply:
Fixed.